doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,495 @@
+* Extend MODULE_TYPE in ModuleBase class with MODULE_TYPE_DPU and MODULE_TYPE_SWITCH to support SmartSwitch
+
+| API | current_usage | switch cpu | dpu | SONiC enhancements |
+| --- | --- | --- | --- | --- |
+| class ModuleBase | # Possible module types MODULE_TYPE_SUPERVISOR MODULE_TYPE_LINE MODULE_TYPE_FABRIC | yes | yes | #new module type MODULE_TYPE_DPU MODULE_TYPE_SWITCH |
+
+#### 3.1.1 ChassisBase class API enhancements
+| API | current_usage | switch cpu | dpu | SONiC enhancements | PD plugin change| comments |
+| --- | --- | --- | --- | --- | --- | --- |
+| get_supervisor_slot(self) | Retrieves the physical-slot of the supervisor-module in the modular chassis. On the supervisor or line-card modules, it will return the physical-slot of the supervisor-module. | na | na | no  | return 0  | chassisd does not like MODULE_INVALID_SLOT smart switch as chassis, where slot is used as follows 0 - switch, 1 - DPU1, 2 - DPU2, .. 8 - DPU8 |
+| get_my_slot(self) | Retrieves the physical-slot of this module in the modular chassis. | na | na | no | return 0/1-8 | 0 - switch, 1 - DPU1, 2 - DPU2, ... 8 - DPU8 |
+| is_modular_chassis(self) | Retrieves whether the sonic instance is part of modular chassis | yes | na | no | return False | smartswitch chassis is fixed, we are extending the modular chassis class to support DPU modules |
+| get_num_modules(self) | Retrieves the number of modules available on this chassis | yes | yes | no | yes | include DPUs |
+| get_all_modules(self) | Retrieves all modules available on this chassis | yes | yes | no | yes | include DPUs |
+| get_module(self, index) | Retrieves module represented by (0-based) index <index> | yes | yes | no | yes | include DPU |
+| get_module_index(self, module_name) | Retrieves module index from the module name | yes | yes | new module name: DPU, SWITCH | yes |  |
+
+#### 3.1.2 ChassisBase class new APIs
+| API | current_usage | switch cpu | dpu | SONiC enhancements | PD plugin change| comments |
+| --- | --- | --- | --- | --- | --- | --- |
+| is_smartswitch(self) | Retrieves whether the sonic instance is part of smartswitch | yes | na | yes | return True | New API for smartswitch |
+| get_module_dpu_port(self, index) | Retrieves the DPU port (internal ASIC port for DPU) represented by DPU index - 1,8 | yes | na | yes | see comments | Platforms that require to overwrite the platform.json file will use this API. See section 3.1.3 below |
+
+#### 3.1.3 NPU to DPU port mapping
+platform.json of NPU/switch will show the NPU port to DPU port mapping. This will be used by services early in the system boot for midplane IP assignment. In this example there are 8 DPUs and ach having a 200G interface.
+```
+"DPUs" : [
+    {
+      "DPU0": {
+                "Ethernet224": "Ethernet0/0"
+       }
+    },
+    {
+       "DPU1": {
+                "Ethernet228": "Ethernet1/0"
+        },
+    },
+    .
+    .
+    {
+       "DPU7": {
+                "Ethernet252": "Ethernet7/0"
+        },
+    },
```

Comment:
@rameshraghupathy the data ports are not used for midplane IP assignments. Please check.

Reply:
This was a mistake.  Fixed it.